compiler/ast/parser.nim

```diff
@@ -220,46 +236,74 @@ proc lineInfo(p: Parser): TLineInfo =
 
 proc indAndComment(p: var Parser, n: ParsedNode, maybeMissEquals = false) =
   if p.realInd:
-    if p.tok.tokType == tkComment: p.rawSkipComment(n)
+    if p.tok.tokType == tkComment:
+      p.rawSkipComment(n)
     elif maybeMissEquals:
       let col = p.bufposPrevious - p.lineStartPrevious
       var info = newLineInfo(p.lex.fileIdx, p.lineNumberPrevious, col)
       p.localError ParserReport(
         kind: rparInvalidIndentation,
         msg: ", maybe you forgot a '=' at $1 ?" % [p.lex.config$info])
-
     else:
       p.localError ParserReport(kind: rparInvalidIndentation)
   else:
     p.skipComment(n)
 
-proc newNode(p: Parser, kind: TNodeKind, token: Token): ParsedNode =
-  ## Create a new leaf parsed node using line information from the parser p
-  return newParsedNode(kind, p.lineInfo, token)
+proc newLitNode(p: Parser, kind: ParsedKindLiteral, lit: Token): ParsedNode =
+  ## Create a leaf literal node
+  newParsedLitNode(kind, p.lex.fileIdx, lit)
 
 proc newNode(
     p: Parser,
-    kind: TNodeKind,
+    kind: ParsedKindWithSons,
+    token: Token,
     subnodes: sink seq[ParsedNode] = @[],
-    info: TLineInfo = p.lineInfo
   ): ParsedNode =
   ## Create a new non-leaf parsed node using line information from the
   ## parser p
-  return newParsedNode(kind, info, subnodes)
+  newParsedNode(kind, p.lex.fileIdx, token, subnodes)
+
+proc currentTokToNode(
+    p: var Parser,
+    kind: ParsedKindWithSons,
+    subnodes: sink seq[ParsedNode] = @[]): ParsedNode =
+  ## Create a new non-leaf parsed node from the current token and advance the
+  ## parser
+  # TODO: come up with a better name before merging
```

Comment:
Maybe `consumeTok`?

Reply:
Thanks, I like `consume`, going to go with a verbose name for now based on that..